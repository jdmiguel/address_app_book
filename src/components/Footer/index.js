import React from 'react';

import { literals } from '../../utils/constants';

const {
  footer: { text, textLink, urlLink },
} = literals;

const Footer = () => (
  <footer>
    <div className="container">
      <p>
        <a href={urlLink}>{textLink}</a>
        {text}
      </p>
    </div>
  </footer>
);

export default Footer;
