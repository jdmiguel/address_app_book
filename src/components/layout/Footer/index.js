import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { literals, settingsIcon, homeIcon } from '../../../utils/constants';

const {
  footer: {
    text,
    link: { settingsText, settingsLink, homeText, homeLink },
    warning,
  },
} = literals;

const Footer = ({ withWarning, withSettings }) => (
  <footer className={withWarning ? 'withWarning' : ''}>
    <div>{withWarning && warning}</div>
    <div className="container">
      <Link to={withSettings ? settingsLink : homeLink}>
        <img src={withSettings ? settingsIcon : homeIcon} alt="icon" />
        {withSettings ? settingsText : homeText}
      </Link>
      {text}
    </div>
  </footer>
);

Footer.propTypes = {
  withWarning: PropTypes.bool,
  withSettings: PropTypes.bool,
};

Footer.defaultProps = {
  withWarning: false,
  withSettings: false,
};

export default Footer;
