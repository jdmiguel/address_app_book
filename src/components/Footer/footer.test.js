import React from 'react';
import { render } from '@testing-library/react';

import Footer from '.';

describe('Component: Footer', () => {
  it('should render', () => {
    const { container } = render(<Footer />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
