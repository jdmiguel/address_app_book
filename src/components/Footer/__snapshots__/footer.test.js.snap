// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Footer should render 1`] = `
<footer>
  <div
    class="container"
  >
    <p>
      <a
        href="https://github.com/jdmiguel/address_app_book"
      >
        Github Repository
      </a>
      This is a coding exercise for Sherpany company
    </p>
  </div>
</footer>
`;
